cypress-tests/cypress/e2e/v2/anonymousNavigation.cy.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  getRandomString,
+  getUserData,
+  validateLoginV2,
+} from "../../support/commands/general";
+import { User } from "../../support/types/user.types";
+import {
+  createProjectIfMissingAPIV2,
+  deleteProjectFromAPIV2,
+  getProjectByNamespaceAPIV2,
+} from "../../support/utils/projectsV2.utils";
+
+const sessionId = ["anonymousNavigation", getRandomString()];
+
+describe("Anonymous users can only access public resources", () => {
+  // Define some project details
+  let randomString: string;
+  let privateProjectName: string;
+  let publicProjectName: string;
+
+  function resetRequiredResources() {
+    randomString = getRandomString();
+    privateProjectName = `anon-private-${randomString}`;
+    publicProjectName = `anon-public-${randomString}`;
+  }
+
+  function browserLogout() {
```

Comment:
Please use `cy.session()` and not this, it is prone to errors. Cypress already supports having separate cookies etc. when giving different session IDs.

Reply:
I restore the session in the `beforeEach` section, and then manually log in and out throughout the test using `browserLogout` and `robustLogin`.
What do you mean by "use cy.session() and not this"?

Reply:
See https://github.com/SwissDataScienceCenter/renku/blob/425dfd82ba1307edfd3d06ddfe273ccac4f37d3c/cypress-tests/cypress/e2e/privateProject.cy.ts#L57

Reply:
> then manually log in and out throughout the test using browserLogout and robustLogin.

This is exactly what is error prone. Prefer using cy.session to switch between users as per the cypress docs https://docs.cypress.io/api/commands/session#Switching-sessions-inside-tests

Reply:
@leafty you mean that instead of logging out through the ui the best thing is to just  switch to another random session. Or a session that essentially is the "empty" session of a user that is not logged in. Right?

I think we can reserve a random session ID or a specific session ID for this purpose. A user that is not logged in and just switch to it.

Reply:
Yes, exactly. Logging out through the UI is slow and I had troubles when I refactored the gateway, which is why the code I pointed earlier was changed to use `cy.session()` to switch between users.

Logging out through the UI should only be done in one or two tests which specifically check the log-in then log-out flow.

Reply:
To be clear, what my comment is about: remove the `browserLogout()` function and use `cy.session()` instead.

Reply:
I switched to using sessions instead of manually logging in/out https://github.com/SwissDataScienceCenter/renku/pull/3906/commits/bcce7bfc38beb88f927da32a6ebc2daddb6060eb

I agree it's a better solution, especially since I ended up logging out/in multiple times, slowing down the test.